virtualizarr/tests/test_readers/test_dmrpp.py

```diff
@@ -20,153 +20,159 @@
+    "nested_groups": textwrap.dedent(
+        """\
+            <?xml version="1.0" encoding="ISO-8859-1"?>
+            <Dataset xmlns="http://xml.opendap.org/ns/DAP/4.0#" xmlns:dmrpp="http://xml.opendap.org/dap/dmrpp/1.0.0#" dapVersion="4.0" dmrVersion="1.0" name="test.dmrpp">
+                <Dimension name="a" size="10"/>
+                <Dimension name="b" size="10"/>
+                <Int32 name="a">
+                    <Dim name="/a"/>
                     <dmrpp:chunks fillValue="-2147483647" byteOrder="LE">
                         <dmrpp:chunk offset="41268" nBytes="4"/>
                     </dmrpp:chunks>
                 </Int32>
-            </Group>
-        </Group>
-    </Dataset>
-    """
-    return DMRParser(root=ET.fromstring(textwrap.dedent(xml_str)))
+                <Int32 name="b">
+                    <Dim name="/b"/>
+                    <dmrpp:chunks fillValue="-2147483647" byteOrder="LE">
+                        <dmrpp:chunk offset="41268" nBytes="4"/>
+                    </dmrpp:chunks>
+                </Int32>
+                <Group name="group1">
+                    <Dimension name="x" size="720"/>
+                    <Dimension name="y" size="1440"/>
+                    <Int32 name="x">
+                        <Dim name="/group1/x"/>
+                        <Attribute name="test" type="String">
+                            <Value>test</Value>
+                        </Attribute>
+                        <dmrpp:chunks fillValue="-2147483647" byteOrder="LE">
+                            <dmrpp:chunk offset="41268" nBytes="4"/>
+                        </dmrpp:chunks>
+                    </Int32>
+                    <Int32 name="y">
+                        <Dim name="/group1/y"/>
+                        <Attribute name="test" type="String">
+                            <Value>test</Value>
+                        </Attribute>
+                        <dmrpp:chunks fillValue="-2147483647" byteOrder="LE">
+                            <dmrpp:chunk offset="41268" nBytes="4"/>
+                        </dmrpp:chunks>
+                    </Int32>
+                    <Group name="group2">
+                        <Int32 name="area">
+                            <Dim name="/group1/x"/>
+                            <Dim name="/group1/y"/>
+                            <dmrpp:chunks fillValue="-2147483647" byteOrder="LE">
+                                <dmrpp:chunk offset="41268" nBytes="4"/>
+                            </dmrpp:chunks>
+                        </Int32>
+                    </Group>
+                </Group>
+            </Dataset>
+            """
+    ),
+}
+
+
+def dmrparser(dmrpp_xml_str: str, tmp_path: Path, filename="test.nc") -> DMRParser:
+    # TODO we should actually create a dmrpp file in a temporary directory
+    # this would avoid the need to pass tmp_path separately
+
+    return DMRParser(
+        root=ET.fromstring(dmrpp_xml_str), data_filepath=tmp_path / filename
+    )
```

Comment:
Turns out right now there is no need for this to be a fixture - fixtures are basically callables that provide something pre-packaged whilst also managing setup and teardown. But with `tmp_path` explicitly passed in at test execution time there is nothing to setup and teardown in here, so it can just be a function instead of a fixture.

Reply:
Sounds good! I just needed them to be separate so I can use different dmrpp XMLs to test different cases. However this dictionary approach makes more sense!